tests/specs/erc20/functional-spec.k

```diff
@@ -0,0 +1,21 @@
+requires "verification.k"
+
+module FUNCTIONAL-SPEC-SYNTAX
+    imports VERIFICATION
+
+    syntax KItem ::= runLemma ( Step ) | doneLemma ( Step )
+ // -------------------------------------------------------
+    rule <k> runLemma(S) => doneLemma(S) ... </k>
+
+    syntax Step ::= Int
+ // -------------------------
+
+endmodule
+
+module FUNCTIONAL-SPEC
+    imports FUNCTIONAL-SPEC-SYNTAX
+
+    claim <k> runLemma(#hashedLocation("Solidity", keccakIntList(OWNER 2 .IntList), SPENDER .IntList))
+           => doneLemma(keccak(#buf(32, SPENDER) ++ #buf(32, keccak(#buf(32, OWNER) ++ #buf(32, 2))))) ... </k>
+      requires #rangeUInt(256, SPENDER) andBool #rangeUInt(256, OWNER)
```

Comment:
`rule keccak(#buf(32, OFFSET2) ++ #buf(32, keccak(#buf(32, OFFSET1) ++ BASE_BA))) => #hashedLocation("Solidity", #asWord(BASE_BA), OFFSET1 OFFSET2 .IntList)` and make sure to include requires clause that makes pre-condition valid (ranges on `OFFSET1`, `OFFSET2`, and `#asWord(BASE_BA)`).